riptide-core/src/main/java/org/zalando/riptide/Requester.java

```diff
@@ -33,45 +33,63 @@
     private final RequestArguments arguments;
     private final Plugin plugin;
 
-    private final Multimap<String, String> query = LinkedHashMultimap.create();
-    private final HttpHeaders headers = new HttpHeaders();
+    private final Multimap<String, String> query;
+    private final HttpHeaders headers;
 
     Requester(final AsyncClientHttpRequestFactory requestFactory, final MessageWorker worker,
             final RequestArguments arguments, final List<Plugin> plugins) {
         this.requestFactory = requestFactory;
         this.worker = worker;
         this.arguments = arguments;
         this.plugin = Plugin.compound(plugins);
+        this.query = LinkedHashMultimap.create();
+        this.headers = new HttpHeaders();
+    }
+
+    private Requester(Requester requester) {
```

Comment:
Instead of copying here, I'd remove this constructor and force every usage of this constructor to use the one from above. That reduces copying to the bare minimum.